src/MOLFiniteDifference/MOL_discretization.jl

```diff
@@ -45,19 +47,29 @@ function SciMLBase.symbolic_discretize(pdesys::ModelingToolkit.PDESystem,discret
     edgemaps = [spacevals[e...] for e in edges]
     initmaps = substitute.(pdesys.depvars,[t=>tspan[1]])
 
-    depvarmaps = reduce(vcat,[substitute.((pdesys.depvars[i],),edgevals) .=> edgevars[i] for i in 1:length(pdesys.depvars)])
+    # Generate map from variable (e.g. u(t,0)) to discretized variable (e.g. u₁(t))
+    subvar(depvar) = substitute.((depvar,),edgevals)
+    depvarmaps = reduce(vcat,[subvar(depvar) .=> edgevars[i] for (i, depvar) in enumerate(pdesys.depvars)])
+    # depvarderivmaps will dictate what to replace the Differential terms with
     if length(nottime) == 1
-        left_weights(j) = DiffEqOperators.calculate_weights(discretization.upwind_order, 0.0, [space[j][1],space[j][2]])
-        right_weights(j) = DiffEqOperators.calculate_weights(discretization.upwind_order, 0.0, [space[j][end-1],space[j][end]])
+        # 1D system
+        left_weights(j) = DiffEqOperators.calculate_weights(discretization.upwind_order, 0.0, space[j][1:2])
+        right_weights(j) = DiffEqOperators.calculate_weights(discretization.upwind_order, 0.0, space[j][end-1:end])
         central_neighbor_idxs(i,j) = [i+CartesianIndex([ifelse(l==j,-1,0) for l in 1:length(nottime)]...),i,i+CartesianIndex([ifelse(l==j,1,0) for l in 1:length(nottime)]...)]
-        derivars = [[dot(left_weights(j),[depvars[j][central_neighbor_idxs(CartesianIndex(2),1)[1:2][2]],depvars[j][central_neighbor_idxs(CartesianIndex(2),1)[1:2][1]]]),
-                    dot(right_weights(j),[depvars[j][central_neighbor_idxs(CartesianIndex(length(space[1])-1),1)[end-1:end][1]],depvars[j][central_neighbor_idxs(CartesianIndex(length(space[1])-1),1)[end-1:end][2]]])]
-                    for j in 1:length(pdesys.depvars)]
-        depvarderivmaps = reduce(vcat,[substitute.((Differential(nottime[j])(pdesys.depvars[i]),),edgevals) .=> derivars[i]
-                                       for i in 1:length(pdesys.depvars) for j in 1:length(nottime)])
+        left_idxs = central_neighbor_idxs(CartesianIndex(2),1)[1:2]
+        right_idxs(j) = central_neighbor_idxs(CartesianIndex(length(space[j])-1),1)[end-1:end]
+        # Constructs symbolic spatially discretized terms of the form e.g. au₂ - bu₁
+        derivars = [[dot(left_weights(j),depvar[left_idxs]), dot(right_weights(j),depvar[right_idxs(j)])]
+                    for (j, depvar) in enumerate(depvars)]
+        # Create list of all the symbolic Differential terms evaluated at boundary e.g. Differential(x)(u(t,0))
+        subderivar(depvar,s) = substitute.((Differential(s)(depvar),),edgevals)
+        # Create map of symbolic Differential terms with symbolic spatially discretized terms
+        depvarderivmaps = reduce(vcat,[subderivar(depvar, s) .=> derivars[i]
+                                       for (i, depvar) in enumerate(pdesys.depvars) for s in nottime])
```

Comment:
this whole loop is only for the 1D case but is written more generally - should it be simplified?

Reply:
I think simplifying it is the wrong direction: we should just finish making it generic to dimensionality. https://github.com/SciML/DiffEqOperators.jl/issues/350 has some starter code in that direction. I just thought it was unproductive to hold up getting some other contributors going to try and finish that, knowing that I have more than a few grant proposals to write.